### PR TITLE
[TensorExpr] Apply transformations randomly to stress test them.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #64774

### CODE REVIEW

**Review by ZolotukhinM — COMMENTED** [review]
Looks good to me (but this is based on my PR so I can't  approve it), I have only nitpicks.
@navahgar, do you have any comments on this?

**Comment by anijain2305** [comment]
Thanks @ZolotukhinM for the reviews. @navahgar Can you please take a look?

File: torch/csrc/jit/tensorexpr/loopnest.cpp [hunk 1]
```diff
@@ -3422,6 +3423,724 @@ bool LoopNest::rfactor(
...
+    SLICE_TAIL,
+    CACHE_ACCESSES,
+    COMPUTE_AT, // better message
+    RFACTOR, // TODO
+    VECTORIZE,
+    VECTORIZE_INNER_LOOPS,
+    ELIMINATE_DEAD_STORES,
+    MAX_TRANSFORM,
+  };
+  bool no_more_inlining = false;
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/loopnest.cpp:3596]: nit: negative flags can be hard to read. Can we use `more_inlining` instead?

**Review by navahgar — COMMENTED** [review]
Mostly LGTM.

Can we add some tests to ensure that nothing breaks and the randomization does the thing we expect? I'm sure it works now, but someone could introduce a breaking change in future which will go unnoticed if we don't have any tests.

**Comment by anijain2305** [comment]
Closing, new PR opened here - https://github.com/pytorch/pytorch/pull/70174

**Pull request closed without merging**